modules/caddyhttp/metrics.go

```diff
@@ -33,7 +35,7 @@ var httpMetrics = struct {
 func initHTTPMetrics() {
 	const ns, sub = "caddy", "http"
 
-	basicLabels := []string{"server", "handler"}
+	basicLabels := []string{"server", "handler", "host"}
```

Comment:
rather than always setting `host` even when `per_host` is false, can you pass through the `perHost` bool and set it conditionally?

```suggestion
	basicLabels := []string{"server", "handler"}
	if perHost {
		basicLabels = append(basicLabels, "host")
	}
```

(same for `httpLabels` below on ln62)